R/pkg/R/DataFrame.R

```diff
@@ -1413,15 +1413,146 @@ setMethod("join",
             dataFrame(sdf)
           })
 
-#' @rdname merge
+#' 
 #' @name merge
 #' @aliases join
+#' @title Merges two data frames
+#' @param x the first data frame to be joined
+#' @param y the second data frame to be joined
+#' @param by a character vector specifying the join columns. If by is not 
+#'   specified, the common column names in \code{x} and \code{y} will be used.
+#' @param by.x a character vector specifying the joining columns for x.
+#' @param by.y a character vector specifying the joining columns for y.
+#' @param all.x a boolean value indicating whether all the rows in x should
+#'              be including in the join
+#' @param all.y a boolean value indicating whether all the rows in y should 
+#'              be including in the join
+#' @param sort a logical argument indicating whether the resulting columns should be sorted              
+#' @details  If all.x and all.y are set to FALSE, a natural join will be returned. If
+#'   all.x is set to TRUE and all.y is set to FALSE, a left outer join will 
+#'   be returned. If all.x is set to FALSE and all.y is set to TRUE, a right 
+#'   outer join will be returned. If all.x and all.y are set to TRUE, a full 
+#'   outer join will be returned. 
+#' @rdname merge
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlContext <- sparkRSQL.init(sc)
+#' df1 <- jsonFile(sqlContext, path)
+#' df2 <- jsonFile(sqlContext, path2)
+#' merge(df1, df2) # Performs a Cartesian
+#' merge(df1, df2, by = "col1") # Performs an inner join based on expression
+#' merge(df1, df2, by.x = "col1", by.y = "col2", all.y = TRUE)
+#' merge(df1, df2, by.x = "col1", by.y = "col2", all.x = TRUE)
+#' merge(df1, df2, by.x = "col1", by.y = "col2", all.x = TRUE, all.y = TRUE)
+#' }
 setMethod("merge",
           signature(x = "DataFrame", y = "DataFrame"),
-          function(x, y, joinExpr = NULL, joinType = NULL, ...) {
-            join(x, y, joinExpr, joinType)
-          })
+          function(x, y, by.x = NULL, by.y = NULL, by = intersect(names(x), names(y)),
+                   all.x = FALSE, all.y = FALSE, all = FALSE,
+                   suffixes = c("_x","_y"), sort = FALSE, ... ) {
 
+            if (missing(x) | missing(y)) {
```

Comment:
R merge() can support merge by row.names, I thinks it makes less sense for SparkR DataFrame, so could we check this case, and throws "not supported"?